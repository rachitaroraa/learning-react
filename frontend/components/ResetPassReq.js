import React, { Component } from "react";
import { Mutation } from "react-apollo";
import gql from "graphql-tag";
import Form from "./styles/Form";
import Error from "./ErrorMessage";

const RESET_REQUEST_MUTATION = gql`
  mutation RESET_REQUEST_MUTATION($email: String!) {
    requestReset(email: $email) {
      message
    }
  }
`;
class ResetPassReq extends Component {
  state = {
    email: ""
  };
  onChangeState = e => {
    this.setState({ [e.target.name]: e.target.value });
  };
  render() {
    return (
      <Mutation mutation={RESET_REQUEST_MUTATION} variables={this.state}>
        {(requestReset, { loading, error, called }) => {
          return (
            <Form
              data-test="form"
              onSubmit={async e => {
                e.preventDefault();
                await requestReset();
                this.setState({ email: "" });
              }}
            >
              <Error error={error} />
              {!error && !loading && called && (
                <p>Success! Check your email for a reset link!</p>
              )}
              <fieldset>
                <label htmlFor="email">
                  Email
                  <input
                    type="email"
                    name="email"
                    value={this.state.email}
                    onChange={this.onChangeState}
                    placeholder="Email"
                    required
                  />
                </label>

                <button type="submit">Reset Password</button>
              </fieldset>
            </Form>
          );
        }}
      </Mutation>
    );
  }
}

export default ResetPassReq;
export { RESET_REQUEST_MUTATION };
