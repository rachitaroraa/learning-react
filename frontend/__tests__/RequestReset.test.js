import { mount } from "enzyme";
import toJSON from "enzyme-to-json";
import { MockedProvider } from "react-apollo/test-utils";
import ResetPassReq, {
  RESET_REQUEST_MUTATION
} from "../components/ResetPassReq";
import wait from "waait";

const mocks = [
  {
    request: {
      query: RESET_REQUEST_MUTATION,
      variables: { email: "rachit4evr@gmail.com" }
    },
    result: {
      data: { requestReset: { message: "success", __typename: "message" } }
    }
  }
];

describe("<ResetPassReq/>", () => {
  it("renders and matches snapshots", async () => {
    const wrapper = mount(
      <MockedProvider mocks={mocks}>
        <ResetPassReq />
      </MockedProvider>
    );
    const form = wrapper.find('form[data-test="form"]');
    expect(toJSON(form)).toMatchSnapshot();
  });
  it("calls the mutation", async () => {
    const wrapper = mount(
      <MockedProvider mocks={mocks}>
        <ResetPassReq />
      </MockedProvider>
    );
    // simulate typing an email
    wrapper.find("input").simulate("change", {
      target: { name: "email", value: "rachit4evr@gmail.com" }
    });
    // submit the form
    wrapper.find("form").simulate("submit");
    await wait();
    wrapper.update();
    expect(wrapper.find("p").text()).toContain(
      "Success! Check your email for a reset link!"
    );
  });
});
