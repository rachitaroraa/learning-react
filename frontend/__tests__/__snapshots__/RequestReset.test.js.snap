// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ResetPassReq/> renders and matches snapshots 1`] = `
<form
  className="Form-sc-1xszr8q-0 krRxky"
  data-test="form"
  onSubmit={[Function]}
>
  <DisplayError
    error={Object {}}
  />
  <fieldset>
    <label
      htmlFor="email"
    >
      Email
      <input
        name="email"
        onChange={[Function]}
        placeholder="Email"
        required={true}
        type="email"
        value=""
      />
    </label>
    <button
      type="submit"
    >
      Reset Password
    </button>
  </fieldset>
</form>
`;
